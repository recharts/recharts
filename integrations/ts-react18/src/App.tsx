import './App.css';
import { Bar, BarChart, Legend, Tooltip, XAxis, YAxis } from 'recharts';

function App() {
  return (
    <>
      <h1>Vite + React 18 + TypeScript + Recharts</h1>
      <BarChart
        width={600}
        height={300}
        data={[
          { name: 'A', value: 100 },
          { name: 'B', value: 200 },
        ]}
      >
        <XAxis dataKey="name" />
        <YAxis />
        <Bar dataKey="value" fill="gold" name="Tooltip value" />
        <Legend />
        <Tooltip />
      </BarChart>
    </>
  );
}

export default App;
