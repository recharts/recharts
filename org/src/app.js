/* eslint-disable no-underscore-dangle */
import React from 'react';
import ReactDOM from 'react-dom';
import { applyRouterMiddleware, browserHistory } from 'react-router';
import { syncHistoryWithStore } from 'react-router-redux';
import { useScroll } from 'react-router-scroll';
import makeRoutes from './routes';
import Root from './containers/Root';
import configureStore from './redux/configureStore';

const initialState = window.__INITIAL_STATE__;
const store = configureStore(initialState, browserHistory);
const history = syncHistoryWithStore(browserHistory, store, {
  selectLocationState: (state) => state.router,
});

const routes = makeRoutes(store);

ReactDOM.render(
  <Root history={history} routes={routes} store={store} render={applyRouterMiddleware(useScroll())} />,
  document.getElementById('app'),
);
