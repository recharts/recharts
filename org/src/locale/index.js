import zhCN from './zh-CN';
import enUS from './en-US';

export default {
  'zh-CN': zhCN,
  'en-US': enUS,
};
