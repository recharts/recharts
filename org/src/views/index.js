import IndexView from './IndexView';
import GuideView from './GuideView';
import APIView from './APIView';
import ExamplesView from './ExamplesView';
import BlogView from './BlogView';
import Storybook from './Storybook';

export { IndexView, GuideView, APIView, ExamplesView, BlogView, Storybook };
