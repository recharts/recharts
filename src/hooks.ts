import type { AxisId } from './state/axisMapSlice';
import { BaseAxisWithScale, selectAxisWithScale } from './state/selectors/axisSelectors';
import { useAppSelector } from './state/hooks';

export const useXAxis = (xAxisId: AxisId): BaseAxisWithScale | undefined =>
  useAppSelector(state => selectAxisWithScale(state, 'xAxis', xAxisId));

export const useYAxis = (yAxisId: AxisId): BaseAxisWithScale | undefined =>
  useAppSelector(state => selectAxisWithScale(state, 'yAxis', yAxisId));
