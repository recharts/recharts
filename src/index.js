import 'ima.js-babel6-polyfill';
import 'core-js/es6/math';
import Surface from './container/Surface';

import CartesianAxis from './component/CartesianAxis';
import CartesianGrid from './component/CartesianGrid';
import Legend from './component/Legend';
import Tooltip from './component/Tooltip';
import PolarGrid from './component/PolarGrid';
import PolarRadiusAxis from './component/PolarRadiusAxis';
import PolarAngleAxis from './component/PolarAngleAxis';
import Brush from './component/Brush';
import ReferenceLine from './component/ReferenceLine';

import Sector from './shape/Sector';
import Curve from './shape/Curve';
import Rectangle from './shape/Rectangle';
import Triangle from './shape/Triangle';
import Polygon from './shape/Polygon';
import Dot from './shape/Dot';

import LineChart from './chartWrapper/LineChart';
import Line from './chart/Line';

import BarChart from './chartWrapper/BarChart';
import Bar from './chart/Bar';

import PieChart from './chartWrapper/PieChart';
import Pie from './chart/Pie';

import TreemapChart from './chartWrapper/TreemapChart';

import RadarChart from './chartWrapper/RadarChart';
import Radar from './chart/Radar';

import ScatterChart from './chartWrapper/ScatterChart';
import Scatter from './chart/Scatter';

import AreaChart from './chartWrapper/AreaChart';
import Area from './chart/Area';

import RadialBarChart from './chartWrapper/RadialBarChart';
import RadialBar from './chart/RadialBar';

import ComposedChart from './chartWrapper/ComposedChart';

import XAxis from './chart/XAxis';
import YAxis from './chart/YAxis';
import ZAxis from './chart/ZAxis';

export {
  // container
  Surface,

  // chart
  Radar,

  // componnet
  CartesianAxis, CartesianGrid,
  Legend, Tooltip, Brush, ReferenceLine,
  PolarGrid, PolarRadiusAxis, PolarAngleAxis,

  // shape
  Sector, Curve, Rectangle, Triangle, Polygon, Dot,

  LineChart, Line,
  AreaChart, Area,
  BarChart, Bar,
  PieChart, Pie,
  ScatterChart, Scatter,
  RadialBarChart, RadialBar,
  TreemapChart,
  RadarChart, Radar,
  ComposedChart,

  XAxis, YAxis, ZAxis,
};
