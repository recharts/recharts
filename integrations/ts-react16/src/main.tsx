import { StrictMode } from 'react';
import { render } from 'react-dom';
import './index.css';
import App from './App.tsx';

render(
  <StrictMode>
    <App />
  </StrictMode>,
  document.getElementById('root')!,
);
