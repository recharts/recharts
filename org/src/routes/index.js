import React, { Component } from 'react';
import { Route, IndexRoute, Redirect, IndexRedirect } from 'react-router';
import _ from 'lodash';
import { IndexView, GuideView, APIView, ExamplesView, BlogView, Storybook } from '../views';
import Frame from '../layouts/Frame';
import Locale from '../locale';

const locales = Object.keys(Locale);

const hasLocale = (pathname) => {
  for (let i = 0, len = locales.length; i < len; i++) {
    if (pathname.indexOf(locales[i]) >= 0) {
      return true;
    }
  }

  return false;
};

const onEnter = (nextState, replace) => {
  const pathname = _.get(nextState, 'location.pathname', '/');

  if (!hasLocale(pathname)) {
    replace({
      pathname: `/en-US${pathname}`,
      query: _.get(nextState, 'location.query', {}),
    });
  }
};

export default (store) => (
  <Route path="/" component={Frame} onEnter={onEnter}>
    <IndexRedirect to="/en-US" />
    <Route path="/*/guide(/:name)" component={GuideView} />
    <Route path="/*/api(/:name)" component={APIView} />
    <Route path="/*/examples(/:name)" component={ExamplesView} />
    <Route path="/*/blog" component={BlogView} />
    <Route path="/*/storybook" component={Storybook} />
    <Route path="*" component={IndexView} />
  </Route>
);
