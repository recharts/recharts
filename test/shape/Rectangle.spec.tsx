import React from 'react';
import each from 'jest-each';
import { render } from '@testing-library/react';

import { Surface, Rectangle } from '../../src';

describe('<Rectangle />', () => {
  each([[5, 10, 8, 15], 5]).it('Should render 1 rectangle in simple Rectangle when radius is %s', radius => {
    const { container } = render(
      <Surface width={400} height={400}>
        <Rectangle x={50} y={50} width={80} height={100} radius={radius} fill="#ff7300" />
      </Surface>,
    );

    expect(container.querySelectorAll('.recharts-rectangle')).toHaveLength(1);
  });

  it('Should render 4 arc when height < 0', () => {
    const { container } = render(
      <Surface width={400} height={400}>
        <Rectangle x={50} y={200} width={80} height={-100} radius={5} fill="#ff7300" />
      </Surface>,
    );

    const rects = container.querySelectorAll('.recharts-rectangle');
    expect(rects).toHaveLength(1);
    expect(rects[0]).toHaveAttribute('d');
    const paths = rects[0].getAttribute('d') || '';
    expect(paths.length - paths.split('A').join('').length).toBe(4);
  });

  it("Shouldn't render anything when height === 0 || width === 0", () => {
    const { container } = render(
      <Surface width={400} height={400}>
        <Rectangle x={50} y={200} width={80} height={0} radius={5} fill="#ff7300" />
        <Rectangle x={50} y={200} width={0} height={30} radius={5} fill="#ff7300" />
      </Surface>,
    );

    expect(container.querySelectorAll('.recharts-rectangle')).toHaveLength(0);
  });

  it("Shouldn't render any path when x, y, width or height is not a number", () => {
    const { container } = render(
      <Surface width={400} height={400}>
        <Rectangle x={'a' as any} y={50} width={80} height={100} fill="#ff7300" />
        <Rectangle x={50} y={'b' as any} width={80} height={100} fill="#ff7300" />
        <Rectangle x={50} y={50} width={'c' as any} height={100} fill="#ff7300" />
        <Rectangle x={50} y={50} width={80} height={'d' as any} fill="#ff7300" />
      </Surface>,
    );

    expect(container.querySelectorAll('.recharts-rectangle')).toHaveLength(0);
  });
});
