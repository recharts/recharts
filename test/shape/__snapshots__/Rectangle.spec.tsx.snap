// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rectangle /> Should render 1 rectangle in simple Rectangle when radius is [ 5, 10, 8, 15 ] 1`] = `
<div>
  <svg
    class="recharts-surface"
    height="400"
    viewBox="0 0 400 400"
    width="400"
  >
    <title />
    <desc />
    <path
      class="recharts-rectangle"
      d="M50,55A 5,5,0,0,1,55,50L 120,50A 10,10,0,0,1,
        130,60L 130,142A 8,8,0,0,1,
        122,150L 65,150A 15,15,0,0,1,
        50,135Z"
      fill="#ff7300"
      height="100"
      radius="5,10,8,15"
      width="80"
      x="50"
      y="50"
    />
  </svg>
</div>
`;

exports[`<Rectangle /> Should render 1 rectangle in simple Rectangle when radius is 5 1`] = `
<div>
  <svg
    class="recharts-surface"
    height="400"
    viewBox="0 0 400 400"
    width="400"
  >
    <title />
    <desc />
    <path
      class="recharts-rectangle"
      d="M 50,55
            A 5,5,0,0,1,55,50
            L 125,50
            A 5,5,0,0,1,130,55
            L 130,145
            A 5,5,0,0,1,125,150
            L 55,150
            A 5,5,0,0,1,50,145 Z"
      fill="#ff7300"
      height="100"
      radius="5"
      width="80"
      x="50"
      y="50"
    />
  </svg>
</div>
`;

exports[`<Rectangle /> Should render 4 arc when height < 0 1`] = `
<div>
  <svg
    class="recharts-surface"
    height="400"
    viewBox="0 0 400 400"
    width="400"
  >
    <title />
    <desc />
    <path
      class="recharts-rectangle"
      d="M 50,195
            A 5,5,0,0,0,55,200
            L 125,200
            A 5,5,0,0,0,130,195
            L 130,105
            A 5,5,0,0,0,125,100
            L 55,100
            A 5,5,0,0,0,50,105 Z"
      fill="#ff7300"
      height="-100"
      radius="5"
      width="80"
      x="50"
      y="200"
    />
  </svg>
</div>
`;

exports[`<Rectangle /> Shouldn't render any path when x, y, width or height is not a number 1`] = `
<div>
  <svg
    class="recharts-surface"
    height="400"
    viewBox="0 0 400 400"
    width="400"
  >
    <title />
    <desc />
  </svg>
</div>
`;

exports[`<Rectangle /> Shouldn't render anything when height === 0 || width === 0 1`] = `
<div>
  <svg
    class="recharts-surface"
    height="400"
    viewBox="0 0 400 400"
    width="400"
  >
    <title />
    <desc />
  </svg>
</div>
`;
